api/project.go

```diff
@@ -158,17 +158,30 @@ func (p *appProjectImpl) addImportsInJson(ignoreError bool, imports ...util.Impo
 	json.Unmarshal([]byte(appDescriptorData), &appDescriptor)
 
 	// list existing imports in JSON to avoid duplicates
-	existingImports := make(map[string]bool)
+	existingImports := make(map[string]string)
 	jsonImports, _ := util.ParseImports(appDescriptor.Imports)
 	for _, e := range jsonImports {
-		existingImports[e.CanonicalImport()] = true
+		existingImports[e.ModulePath()] = e.CanonicalImport()
 	}
 
 	for _, i := range imports {
-		if _, ok := existingImports[i.CanonicalImport()]; !ok {
-			appDescriptor.Imports = append(appDescriptor.Imports, i.CanonicalImport())
+		val, ok := existingImports[i.ModulePath()]
+		if !ok {
+			//appDescriptor.Imports = append(appDescriptor.Imports, i.CanonicalImport())
+			existingImports[i.ModulePath()] = i.CanonicalImport()
+		} else {
+			if i.CanonicalImport() != val {
+				delete(existingImports, val)
+				existingImports[i.ModulePath()] = i.CanonicalImport()
```

Comment:
If there is an alias in the existing import we should keep it not to break the ```"ref": "#alias"``` references in other parts of *flogo.json*.

```suggestion
				alias := i.Alias()
				if val.Alias() != "" && i.Alias() == "" {
					alias = val.Alias()
				}
				existingImports[i.GoImportPath()] = util.NewFlogoImport(i.ModulePath(), i.RelativeImportPath(), i.Version(), alias)
```